interfaces/builtin/system_files_test.go

```diff
@@ -0,0 +1,181 @@
+		Snap:      &snap.Info{SuggestedName: "core", Type: snap.TypeOS},
+		Name:      "system-files",
+		Interface: "system-files",
+	}
+	s.slot = interfaces.NewConnectedSlot(s.slotInfo, nil, nil)
+	plugSnap := snaptest.MockInfo(c, mockPlugSnapInfo, nil)
+	s.plugInfo = plugSnap.Plugs["system-files"]
+	s.plug = interfaces.NewConnectedPlug(s.plugInfo, nil, nil)
+}
+
+func (s *systemFilesInterfaceSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "system-files")
+}
+
+func (s *systemFilesInterfaceSuite) TestConnectedPlugAppArmor(c *C) {
+	apparmorSpec := &apparmor.Specification{}
+	err := apparmorSpec.AddConnectedPlug(s.iface, s.plug, s.slot)
+	c.Assert(err, IsNil)
+	c.Assert(apparmorSpec.SecurityTags(), DeepEquals, []string{"snap.other.app"})
+	c.Check(apparmorSpec.SnippetForTag("snap.other.app"), Equals, `
+# Description: Can access specific system files or directories.
+# This is restricted because it gives file access to arbitrary locations.
+"/etc/read-dir2{,/,/**}" rk,
+"/etc/read-file2{,/,/**}" rk,
+"/etc/write-dir2{,/,/**}" rwkl,
+"/etc/write-file2{,/,/**}" rwkl,
+`)
+}
+
+func (s *systemFilesInterfaceSuite) TestSanitizeSlot(c *C) {
+	c.Assert(interfaces.BeforePrepareSlot(s.iface, s.slotInfo), IsNil)
+	slot := &snap.SlotInfo{
+		Snap:      &snap.Info{SuggestedName: "some-snap"},
+		Name:      "system-files",
+		Interface: "system-files",
+	}
+	c.Assert(interfaces.BeforePrepareSlot(s.iface, slot), ErrorMatches,
+		"system-files slots are reserved for the core snap")
+}
+
+func (s *systemFilesInterfaceSuite) TestSanitizePlug(c *C) {
+	c.Assert(interfaces.BeforePreparePlug(s.iface, s.plugInfo), IsNil)
+}
+
+func (s *systemFilesInterfaceSuite) TestSanitizePlugHappy(c *C) {
+	const mockSnapYaml = `name: system-files-plug-snap
+version: 1.0
+plugs:
+ system-files:
+  read: ["/etc/file1"]
+  write: ["/etc/dir1"]
+`
+	info := snaptest.MockInfo(c, mockSnapYaml, nil)
+	plug := info.Plugs["system-files"]
+	c.Assert(interfaces.BeforePreparePlug(s.iface, plug), IsNil)
+}
+
+func (s *systemFilesInterfaceSuite) TestSanitizePlugUnhappy(c *C) {
+	const mockSnapYaml = `name: system-files-plug-snap
+version: 1.0
+plugs:
+ system-files:
+  $t
+`
+	errPrefix := `cannot add system-files plug: `
+	var testCases = []struct {
+		inp    string
+		errStr string
+	}{
+		{`read: ""`, `"read" must be a list of strings`},
+		{`read: [ 123 ]`, `"read" must be a list of strings`},
+		{`read: [ "/foo/./bar" ]`, `"/foo/./bar" must be clean`},
+		{`read: [ "../foo" ]`, `"../foo" must start with "/"`},
+		{`read: [ "/foo[" ]`, `"/foo\[" contains a reserved apparmor char from .*`},
+		{`write: ""`, `"write" must be a list of strings`},
+		{`write: bar`, `"write" must be a list of strings`},
+		{`read: [ "~/foo" ]`, `"~/foo" contains invalid "~"`},
+		{`read: [ "/foo/~/foo" ]`, `"/foo/~/foo" contains invalid "~"`},
+		{`read: [ "/foo/../foo" ]`, `"/foo/../foo" must be clean`},
+		{`read: [ "/home/$HOME/foo" ]`, `\$HOME cannot be used in "/home/\$HOME/foo"`},
```

Comment:
This is an interesting test case. We could disallow '/home' as a prefix, but we would always be chasing configurations where the user home directories are somewhere else. I think you are right to not second guess the snap and let it be a matter of snap declaration store policies on what we allow.